### PR TITLE
Fix/resources

### PR DESCRIPTION
**Pull request by szuecs**
do not overcommit memory and cput settings and use guaranteed prio class for all system components

### CODE REVIEW

File: cluster/manifests/kube2iam/daemonset.yaml [hunk 1]
```diff
@@ -56,8 +56,8 @@ spec:
...
         resources:
           limits:
-            cpu: 100m
-            memory: 100Mi
+            cpu: 25m
+            memory: 50Mi
           requests:
             cpu: 25m
-            memory: 25Mi
+            memory: 50Mi
```
Comment (aermakov-zalando) [inline cluster/manifests/kube2iam/daemonset.yaml]: This won't work unless it's `OnDelete`. Nodes will end up having no `kube2iam` pod at all.

**Comment by aermakov-zalando** [comment]
Increasing the requests will just end up in nodes not running the pod at all, unless you change it to `OnDelete`.

**Comment by aermakov-zalando** [comment]
👎

File: cluster/manifests/kube2iam/daemonset.yaml (hunk 1 above)
Comment (szuecs) [reply cluster/manifests/kube2iam/daemonset.yaml]: fixed @aermakov-zalando

Comment (aermakov-zalando) [reply cluster/manifests/kube2iam/daemonset.yaml]: Please fix the rest (requests are increased in other daemonsets as well).

Comment (szuecs) [reply cluster/manifests/kube2iam/daemonset.yaml]: done, would be nice if you would tell the next time this upfront

Comment (aermakov-zalando) [reply cluster/manifests/kube2iam/daemonset.yaml]: sorry!

Comment (aermakov-zalando) [reply cluster/manifests/kube2iam/daemonset.yaml]: have you verified that it won't cause issues, btw?

File: cluster/manifests/logging-agent/daemonset.yaml [hunk 2]
```diff
@@ -112,11 +112,11 @@ spec:
 
         resources:
           limits:
-            cpu: 200m
-            memory: 200Mi
+            cpu: 100m
+            memory: 150Mi
```
Comment (aermakov-zalando) [inline cluster/manifests/logging-agent/daemonset.yaml]: This definitely won't work, I've seen in crashing in a couple of clusters with the existing `200Mi` limit. There's another PR (https://github.com/zalando-incubator/kubernetes-on-aws/pull/1486) to increase it instead.

**Comment by aermakov-zalando** [comment]
Where do the new limits come from? Are we sure that stuff won't start randomly dying under load?

File: cluster/manifests/logging-agent/daemonset.yaml (hunk 2 above)
Comment (szuecs) [reply cluster/manifests/logging-agent/daemonset.yaml]: I adjusted the resource to the ones from the other PR and closed the other PR

**Comment by szuecs** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**